Update dynamic_layers.py and upscaling utilities.py

jbwhitco opened the pull request:
Changed line 209 to have the variable 'elements' formed based on just the basename of the UAVSAR file, such that element[1] is always the date string of the UAVSAR file.

jbwhitco committed:
Update dynamic_layers.py

Changed line 209 to have the variable 'elements' formed based on just the basename of the UAVSAR file, such that element[1] is always the date string of the UAVSAR file.

jbwhitco committed:
Add files via upload

jbwhitco committed:
Delete upscaling_utilities.py

File needs to be moved into the soilscape_upscaling folder

jbwhitco committed:
Delete upscaling_utilities.py

Deleting xml version so it can be replaced by python-only version.

jbwhitco committed:
Add files via upload

jbwhitco commented:
Hopefully this pull request includes the following two changes:
1) Minor bug correction on line 209 of dynamic layers.py.
2) Replace upscaling_utilities.py with a python-only (i.e., no xml) version.

danclewley commented on soilscape_upscaling/upscaling_utilities.py:
```diff
@@ -5,15 +5,14 @@
 
 General purpose utilities
-
-This file is licensed under the GPL v3 Licence. A copy of this
```
Need to keep licence text in header.

danclewley commented on soilscape_upscaling/upscaling_utilities.py:
```diff
@@ -29,92 +28,101 @@ def get_gdal_format(file_name):
-    Colour image using RSGISLib XML interface
 
-    FIXME: This function should be tidied up possibly to use
+def colour_SM_image(inimage, colorimage, max_value=0.5, band=1):
```
Would be better to use the English spelling `colourimage` for consistency.

danclewley commented on soilscape_upscaling/upscaling_utilities.py:
```diff
@@ -29,92 +28,101 @@ def get_gdal_format(file_name):
+        print('expression: ',expression)
+        imagecalc.bandMath(colorimage, expression, gdalformat, datatype, bandDefns)       
+        for i in range(11):
+            os.system('rm '+color[i])
```
Better to use `os.remove` here, should also work under Windows. Alternatively could use a temp directory (`tempfile.mkdtemp`) for the individual colour bands and remove the entire thing using `shutil.rmtree` which would also take care of any  .aux.xml ancillary files.

danclewley reviewed (changes requested):
Thanks for the updates. I've made some minor changes it would be good to make before merging. As some stage I'll try and update so it doesn't require creating 11 new images before colouring.

jbwhitco committed:
Delete upscaling_utilities.py

To be replaced with a version that corrects the header and variable names and streamlines the removal of temporary files.

jbwhitco committed:
Add files via upload

New version that corrects header and variable names and streamlines the removal of temporary files.

jbwhitco committed:
Delete upscaling_utilities_noxml_r1.py

Need to rename file to upscaling_utilities.py

jbwhitco committed:
Add files via upload

Just renamed file to be recognized by calling script.

jbwhitco commented:
Corrected header text and variable names and streamlined removal of temporary files.
